service-base/src/main/java/org/eclipse/hono/service/registration/RegistrationHttpEndpoint.java

```diff
@@ -75,33 +75,30 @@ public String getName() {
     @Override
     public void addRoutes(final Router router) {
 
+        final String pathWithTenant = String.format("/%s/:%s", RegistrationConstants.REGISTRATION_ENDPOINT, PARAM_TENANT);
         // ADD device registration
-        router.route(HttpMethod.POST, String.format("/%s/:%s", RegistrationConstants.REGISTRATION_ENDPOINT, PARAM_TENANT))
-            .consumes(HttpEndpointUtils.CONTENT_TYPE_JSON)
-            .handler(this::doRegisterDeviceJson);
-        router.route(HttpMethod.POST, String.format("/%s/:%s", RegistrationConstants.REGISTRATION_ENDPOINT, PARAM_TENANT))
-            .consumes(HttpHeaders.APPLICATION_X_WWW_FORM_URLENCODED.toString())
-            .handler(this::doRegisterDeviceForm);
-        router.route(HttpMethod.POST, String.format("/%s/*/*", RegistrationConstants.REGISTRATION_ENDPOINT))
```

Comment:
Didn't work because asterisk * can only be used at the end of the path.
Alternatives: using routeWithRegex() or using placeholders - as done with the other routes and as now done in the PR.